stylix/mk-target.nix

```
@@ -225,20 +225,24 @@ let
       mkConfig = fn: fn (getStylixAttrs fn);
 
       # Safeguard configuration functions when any of their arguments is
-      # disabled, while non-function configurations are unguarded.
+      # disabled.
       mkConditionalConfig =
-        c:
-        if builtins.isFunction c then
-          let
-            allAttrsNonNull = lib.pipe c [
-              getStylixAttrs
-              builtins.attrValues
-              (builtins.all (attr: attr != null))
-            ];
-          in
-          lib.mkIf allAttrsNonNull (mkConfig c)
+        let
+          allAttrsNonNull = lib.pipe config [
+            getStylixAttrs
+            builtins.attrValues
+            (builtins.all (attr: attr != null))
+          ];
+        in
+        config:
+        if builtins.isFunction config then
+          lib.mkIf allAttrsNonNull (mkConfig config)
+
+        else if builtins.isPath config then
+          lib.mkIf allAttrsNonNull (mkConfig (import config))
```

Since `configElements` cannot define options, I assume using `lib.modules.importApply` would have no actual effect:

```suggestion
          lib.mkIf allAttrsNonNull (mkConfig (lib.modules.importApply config))
```

Maybe I missed a use case where using `lib.modules.importApply` is better. Or maybe we should just use it to be safe?

---

> Maybe I missed a use case where using `lib.modules.importApply` is better. Or maybe we should just use it to be safe?

It seems like `lib.modules.importApply` cannot be trivially used in this case, if that is even possible.